playwright-e2e/dsm/pages/tissue-information-page/tissue-information-page.ts

```diff
@@ -0,0 +1,294 @@
+import {expect, Locator, Page} from '@playwright/test';
+import {DynamicFieldsEnum, ProblemWithTissueEnum, TissueInformationEnum} from './enums/tissue-information-enum';
+import DatePicker from '../../component/date-picker';
+import {waitForResponse} from '../../../utils/test-utils';
+import {FillDate} from './interfaces/tissue-information-interfaces';
+import Select from '../../../dss/component/select';
+import TextArea from '../../../dss/component/textarea';
+import Tissue from '../../component/tissue';
+import Checkbox from '../../../dss/component/checkbox';
+import Button from '../../../dss/component/button';
+import Input from '../../../dss/component/input';
+
+
+export default class TissueInformationPage {
+  private readonly PAGE_TITLE = 'Tissue Request';
+
+  constructor(private readonly page: Page) {}
+
+  public async tissue(index = 0): Promise<Tissue> {
+    await this.tissuesCountCheck(index);
+    return new Tissue(this.page, index);
+  }
+
+  public async backToParticipantPage(): Promise<void> {
+    await this.page.getByText('<< back to Participant Page').click();
+    await this.page.waitForLoadState('load');
+  }
+
+  public async backToList(): Promise<void> {
+    await this.page.getByText('<< back to List').click();
+    await this.page.waitForLoadState('load');
+  }
+
+  public async addTissue(): Promise<Tissue> {
+    const addTissueBtnLocator = this.addTissueButton;
+    await expect(addTissueBtnLocator, `Add Tissue button is not visible`).toBeVisible();
+    const button = new Button(this.page, {root: addTissueBtnLocator});
+    const isDisabled = await button.isDisabled();
+
+    if (isDisabled) {
+      throw new Error('Add tissue button is disabled');
+    }
+
+    await button.click();
+    const tissuesCount = await this.tissuesCount;
+    return this.tissue(tissuesCount - 1);
+  }
+
+  public async deleteTissueByIndex(tissueIndex: number): Promise<void> {
+    await this.tissuesCountCheck(tissueIndex);
+    const deleteTissueBtnLocator = this.deleteTissueButton(tissueIndex);
+    await expect(deleteTissueBtnLocator, `Tissue at index '${tissueIndex}' is not visible`).toBeVisible();
+
+    const button = new Button(this.page, {root: deleteTissueBtnLocator});
+    const isDisabled = await button.isDisabled();
+
+    if (isDisabled) {
+      throw new Error('Delete tissue button is disabled');
+    }
+
+    await button.click();
+  }
+
+  public async getParticipantInformation(name: TissueInformationEnum): Promise<string> {
+    const participantInformation = this.participantInformation(name);
+    await expect(participantInformation, `Data can't be found on - ${name}`).toBeVisible();
+
+    const data = await participantInformation.textContent();
+    return data?.trim() as string;
+  }
+
+  public async fillFaxSentDates(date1: FillDate, date2?: FillDate, date3?: FillDate): Promise<void> {
```

Comment:
I think you want use rest parameter syntax: `fillFaxSentDates(...date: FillDate)`

Reply:
I believe I'd like to keep it as it is since it appears more readable to me. With just three parameters to define, there's no need for looping through them. However, I do acknowledge that the rest parameter is a fantastic feature in JavaScript.